### PR TITLE
Split client and provider to 2 different interfaces

### PR DESCRIPTION
**Pull request by edwardrf**
Rename type `Provider` to `ProviderID` and create `Provider` interface which takes all the deployment related features of the old `Client` interface, which currently only takes all the GrpcClient methods.

Test showing AI debugging logic still works correctly:
```
crasher$ defang compose up
 ! Using Defang playground, but AWS environment variables were detected; did you forget --provider=aws or DEFANG_PROVIDER=aws?
 * Using Defang Playground; consider using BYOC (https://s.defang.io/byoc)
 ! service "crasher": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 * Packaging the project files for crasher at /home/edw/defang/tmp/crasher
 ! The build context is more than 10MiB; use --debug or create .dockerignore to exclude caches and build artifacts
 * Uploading the project files for crasher
 * Monitor your services' status in the defang portal
   - https://portal.defang.dev/service/crasher
 * Tailing logs for deployment ID 216073bv0mmz ; press Ctrl+C to detach:
 * Showing only build logs and runtime errors. Press V to toggle verbose mode.
2024-10-29T09:47:56.128486-07:00 cd Update started for stack edwardrf-prod1
2024-10-29T09:48:00.876843-07:00 cd  ** Building image for "crasher"...
2024-10-29T09:48:03.339894-07:00 cd  ** Build succeeded for "crasher"
2024-10-29T09:48:04.351981-07:00 cd  ** Updated service "crasher" to revision 3
2024-10-29T09:48:09.256157-07:00 cd Update succeeded in 13.170014949s ; provisioning...
2024-10-29T09:48:43.923673-07:00 crasher exec ./main: exec format error
2024-10-29T09:49:38.957524-07:00 crasher exec ./main: exec format error
2024-10-29T09:50:52.029647-07:00 crasher exec ./main: exec format error
 ! deployment failed for service "crasher"
? Would you like to debug the deployment with AI? Yes
```

### CODE REVIEW

File: src/cmd/cli/command/compose.go [hunk 1]
```diff
@@ -166,6 +166,7 @@ func makeComposeUpCmd() *cobra.Command {
 				if errors.As(context.Cause(tailCtx), &errDeploymentFailed) {
 					// Tail got canceled because of deployment failure: prompt to show the debugger
 					term.Warn(errDeploymentFailed)
+					// FIXME: THis should nolonger be working
```
Comment (lionello) [inline src/cmd/cli/command/compose.go:169]: should probably move everything that's inside the `if` into a method?

File: src/pkg/cli/client/client.go [hunk 2]
```diff
@@ -20,7 +19,7 @@ type ProjectLoader interface {
 	LoadProject(context.Context) (*composeTypes.Project, error)
 }
 
-type FabricClient interface {
+type Client interface {
```
Comment (lionello) [inline src/pkg/cli/client/client.go:22]: I'd keep this as `FabricClient`

File: src/pkg/cli/client/playground.go [hunk 3]
```diff
@@ -135,6 +135,16 @@ func (g PlaygroundClient) LoadProjectName(ctx context.Context) (string, error) {
...
 	g.projectName = projectName
 }
+
+func (g *PlaygroundProvider) AccountInfo(ctx context.Context) (AccountInfo, error) {
+	return PlaygroundAccountInfo{}, nil
+}
+
+type PlaygroundAccountInfo struct{}
+
+func (g PlaygroundAccountInfo) AccountID() string { return "" }
```
Comment (lionello) [inline src/pkg/cli/client/playground.go:148]: tenantId?

File: src/pkg/cli/client/playground.go [hunk 4]
```diff
@@ -135,6 +135,16 @@ func (g PlaygroundClient) LoadProjectName(ctx context.Context) (string, error) {
...
+func (g PlaygroundAccountInfo) Region() string    { return "" }
```
Comment (lionello) [inline src/pkg/cli/client/playground.go:149]:
```suggestion
func (g PlaygroundAccountInfo) Region() string    { return "us-west-2" }
```

File: src/pkg/cli/tail.go [hunk 5]
```diff
@@ -247,7 +247,7 @@ func tail(ctx context.Context, client client.Client, params TailOptions) error {
 								modeStr += ". I like the way you work it, no verbosity."
 							}
 							term.Info("Verbose mode", modeStr)
-							go client.Track("Verbose Toggled", P{"verbose", verbose}, P{"toggleCount", toggleCount})
+							// go client.Track("Verbose Toggled", P{"verbose", verbose}, P{"toggleCount", toggleCount}) // FIXME: Do we need to pass client around to track?
```
Comment (lionello) [inline src/pkg/cli/tail.go:250]: restore?

**Review by lionello — APPROVED** [review]

File: src/pkg/cli/client/playground.go (hunk 4 above)
Comment (edwardrf) [reply src/pkg/cli/client/playground.go:149]: Good idea

File: src/pkg/cli/tail.go (hunk 5 above)
Comment (edwardrf) [reply src/pkg/cli/tail.go]: I'll have to tackle it in the next PR, if I use command.Track it becomes a circular depencency, if I pass the client all the way down from the commands, it touches too many function signature that would be restored once I have the track package restructuring done. I've added a fixme for future reference.

**Pull request merged**